torchrl/envs/transforms/transforms.py

```diff
@@ -3893,28 +3893,59 @@ class FrameSkipTransform(Transform):
     Args:
         frame_skip (int, optional): a positive integer representing the number
             of frames during which the same action must be applied.
+        action_interp (bool, optional): whether to perform action interpolation over frame_skip steps.
 
     """
 
-    def __init__(self, frame_skip: int = 1):
-        super().__init__()
+    def __init__(self, frame_skip: int = 1, action_interp: bool = False):
+        super().__init()
         if frame_skip < 1:
             raise ValueError("frame_skip should have a value greater or equal to one.")
         self.frame_skip = frame_skip
+        self.action_interp = action_interp
+        self.action_interp_buffer = None
 
     def _step(
         self, tensordict: TensorDictBase, next_tensordict: TensorDictBase
     ) -> TensorDictBase:
         parent = self.parent
         if parent is None:
             raise RuntimeError("parent not found for FrameSkipTransform")
+
+        if self.action_interp:
+            action_key = "_action"
+            current_action = tensordict.get(action_key)
+            next_action = next_tensordict.get(action_key)
+            if self.action_interp_buffer is not None:
+                interpolated_actions = self._linear_interpolation(
+                    self.action_interp_buffer, next_action, self.frame_skip
+                )
+                self.action_interp_buffer = next_action
+            else:
+                interpolated_actions = [current_action] * (self.frame_skip - 1)
+                self.action_interp_buffer = next_action
+            next_tensordict.set(action_key, interpolated_actions)
+
         reward_key = parent.reward_key
         reward = next_tensordict.get(reward_key)
         for _ in range(self.frame_skip - 1):
             next_tensordict = parent._step(tensordict)
             reward = reward + next_tensordict.get(reward_key)
+
         return next_tensordict.set(reward_key, reward)
 
+    def _linear_interpolation(
+        self, start_action: Tensor, end_action: Tensor, num_steps: int
+    ) -> List[Tensor]:
+        if num_steps <= 0:
+            return [start_action]
+        interpolation_steps = [start_action]
+        for step in range(1, num_steps):
+            alpha = step / num_steps
+            interpolated_action = start_action + alpha * (end_action - start_action)
+            interpolation_steps.append(interpolated_action)
```

Comment:
I'm not sure I see what this block is doing, can you give an example?
It'd be cool to see (eg in the tests) what you would expect to get from it

Reply:
Ok, so my reasoning here was for let's say `num_steps = 5`, we have:

At `step = 1`, `alpha = 1/5` and `interpolated_action` is calculated as follows:
`interpolated_action=start_action+(1/5)*(end_action - start_action)`

Similarly for each step (like for the 4th step, `alpha=4/5` ) and `interpolated_action` is calculated as follows:
`interpolated_action=start_action+(4/5)*(end_action - start_action)`

And finally, at `step = 5`, `alpha = 5/5` (which is 1), and `interpolated_action` becomes :
`interpolated_action=end_action`

So like the`interpolation_steps` list contains the sequence of actions that go from from `start_action` to `end_action` over the specified number of steps, which in this example is 5. 
I felt that this made sense for moving from `start_action` to `end_action` , for smooth action transitions over multiple steps

Reply:
I don't know if this makes sense, but I felt that this is how i have to do it

Reply:
Ok got it, do you mind if I try to do a vectorized version of this once you have written tests? I think we can make it faster

Reply:
Suree! please do it!
I'll send another round of commits by the next 3ish hours
Also please could you answer what 

> would it make sense to store this contiguously (with a torch.stack around it)?

This means?